sdk/sdk-api/src/main/java/mil/emp3/api/utils/kml/KMLExportThread.java

```diff
@@ -530,6 +545,46 @@ public void serializeGeometry(final XmlSerializer xmlSerializer) throws IOExcept
         return oArray;
     }
 
+    private static String convertDocumentToString(final Document doc) {
+        final TransformerFactory tf = TransformerFactory.newInstance();
+        try {
+            final Transformer transformer = tf.newTransformer();
+            // below code to remove XML declaration
+            transformer.setOutputProperty(OutputKeys.OMIT_XML_DECLARATION, "yes");
+            final StringWriter writer = new StringWriter();
+            transformer.transform(new DOMSource(doc), new StreamResult(writer));
+            return writer.getBuffer().toString();
+        } catch (final TransformerException ex) {
+            ex.printStackTrace();
+        }
+        return null;
+    }
+    /**
+     * Circles, ellipses, squares, and rectangles are represented by a path and a polygon
+     * in kml.  Therefore we parse out the path so that we only return the polygon part
+     * in the kml that emp exports.
+     */
+    private String parsePolygonFromMultipleFeatureKML(final String kml) {
+        try {
+            final DocumentBuilderFactory dbFactory = DocumentBuilderFactory.newInstance();
+            final DocumentBuilder dBuilder = dbFactory.newDocumentBuilder();
+            final Document doc = dBuilder.parse(new InputSource(new StringReader(kml)));
```

Comment:
StringReader is also a resource that needs to be closed in order to not have a memory leak.  Please also put this in a Try with resources.